### PR TITLE
Improve layout article

### PR DESCRIPTION
**Pull request by cadamini**
Closes https://github.com/utkarsh006/Eazy-Android/issues/46

I only wanted to fix the h1 and h2 heading formatting to make it consistent (h1 on top, etc)

But then I spotted some more areas for improvement in the article: 

- fixed text and formatting issues
- removed duplication
- fixed inconsistencies. 
- converted the parameter list at the end to a table.

Please review carefully :)

Note: I haven't checked the texts in the layout example section(s) though.

### CODE REVIEW

File: Articles/9_Layout_and_their_types.md [hunk 1]
```diff
@@ -1,20 +1,15 @@
...
-The <b>ViewGroup is a subclass of View</b> and provides invisible container that hold other Views or other ViewGroups and define their layout properties.
-
-At third level we have different <b> layouts which are subclasses of ViewGroup class</b> and a typical layout defines the visual structure for an Android user interface and can be created either at run time using View/ViewGroup objects or you can declare your layout using simple XML file main_layout.xml which is located in the res/layout folder of your project.
-
-# Layout
-As explained above, designing gets done in these stages:
-- you have <b>view</b> that is base class for widgets like button, textView etc...
-- you have group of those views, an invisible container that holds other views which can be termed as <b>ViewGroup</b> like RecyclerView, ListView etc...
-- We have <b>layouts</b> like LinearLayout, RelativeLayout etc... that hold views and viewgroups. 
-
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:13]: reviewer note: both parts were basically duplicated

File: Articles/9_Layout_and_their_types.md [hunk 2]
```diff
@@ -26,10 +21,9 @@ Lets understand the graph:
                                                                
 
 </pre>
-<hr>
 
-Typical syntax is as follows:
-<br><hr>
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:32]: html tags not needed in GitHub markdown

File: Articles/9_Layout_and_their_types.md [hunk 3]
```diff
@@ -51,26 +45,27 @@ Typical syntax is as follows:
...
+- Create your layouts at run time using View/ViewGroup objects or declare your layout using an XML file main_layout.xml which is located in the res/layout folder of your project.
 
-# Types of Layout
-There are numerous Layouts provided by Android which you will use in almost all the Android applications to provide different view, look and feel.
-Let's look at some of the layouts:
+## Different layout types
+
+There are numerous Layouts provided by Android which you will use in almost all the Android applications to provide different view, look and feel. Let's look at some of the layouts below.
+
+## GridView: ViewGroup to display items in a two-dimensional and scrollable grid
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:67]:
```suggestion
### GridView: ViewGroup to display items in a two-dimensional and scrollable grid
```

File: Articles/9_Layout_and_their_types.md [hunk 4]
```diff
@@ -102,11 +97,11 @@ Lets consider UI of Myntra's fashion wear page. We can see a scrollable 2D list
                               |___________________________|
 
 </pre>
-<br>
 
-- ##  ListView: ListView is a view group that displays a list of scrollable items.
+## ListView: ViewGroup to display a list of scrollable items
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:101]:
```suggestion
### ListView: ViewGroup to display a list of scrollable items
```

File: Articles/9_Layout_and_their_types.md [hunk 5]
```diff
@@ -135,12 +130,11 @@ ListView as the name suggests, displays items in list format that can be scrolle
                               |_________Listiew___________|
 
 </pre>
-<br>
 
-- ##  FrameLayout: The FrameLayout is a placeholder on screen that you can use to display a single view.
+##  FrameLayout: Placeholder on screen that you can use to display a single view
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:134]:
```suggestion
### FrameLayout: Placeholder on the screen that you can use to display a single view
```

File: Articles/9_Layout_and_their_types.md [hunk 6]
```diff
@@ -169,12 +163,11 @@ An example would be, if you want to have lottie animations in a particular part
                               |___________________________|
 
 </pre>
-<br>
 
-- ##  AbsoluteLayout: AbsoluteLayout enables you to specify the exact location of its children.
+##  AbsoluteLayout: Specify the exact location of its children
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:167]:
```suggestion
### AbsoluteLayout: Specify the exact location of its children
```

File: Articles/9_Layout_and_their_types.md [hunk 7]
```diff
@@ -203,12 +196,11 @@ In this layout, the exact location is specified of the child items present there
                               |___________________________|
 
 </pre>
-<br>
 
-- ##  TableLayout: TableLayout is a view that groups views into rows and columns.
+## TableLayout: View that groups views into rows and columns
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:200]:
```suggestion
### TableLayout: View that groups views into rows and columns
```

File: Articles/9_Layout_and_their_types.md [hunk 8]
```diff
@@ -237,14 +229,13 @@ If you wish to form a table and display it on screen just as we have tables in h
                               |___________________________|
 
 </pre>
-<br>
 
+## RelativeLayout: ViewGroup that displays child views in relative positions
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:233]:
```suggestion
### RelativeLayout: ViewGroup that displays child views in relative positions
```

File: Articles/9_Layout_and_their_types.md [hunk 9]
```diff
@@ -278,12 +269,11 @@ Well, this can be precisely made possible using RelativeLayout wherein you can d
                               |___________________________|
 
 </pre>
-<br>
 
-- ##  LinearLayout: LinearLayout is a view group that aligns all children in a single direction, vertically or horizontally.
+## LinearLayout: ViewGroup that aligns all children in a single direction (vertically or horizontally)
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:273]:
```suggestion
### LinearLayout: ViewGroup that aligns all children in a single direction (vertically or horizontally)
```

File: Articles/9_Layout_and_their_types.md [hunk 10]
```diff
@@ -313,18 +303,16 @@ Unlike RelativeLayout, LinearLayout places items in single direction be it horiz
                               |________LinearLayout_______|
 
 </pre>
-<br>
 
- - ## Constraint Layout: Views in this layout have are alinged with a invisible constraint
- Consider constraints as springs attached to all the 4 sides of your view. This binds the view from moving to random place (defined in pixels) while in runtime as the application will run on multiple devices of different shape and forms. The spring mechanism (just to explain) helps to fix the view at one place.
+## Constraint Layout: Views in this layout have are alinged with a invisible constraint
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:307]:
```suggestion
### ConstraintLayout: Views in this layout are aligned with an invisible constraint
```

File: Articles/9_Layout_and_their_types.md [hunk 11]
```diff
@@ -1,20 +1,15 @@
...
-<br><hr>
+# Layouts - types, attributes, and properties
+
+The basic building block for user interfaces is a **View object** that is responsible for drawing and event handling. 
+
+Designing is done in three stages:
+
+- **View** is the base class for widgets, which are used to create interactive UI components like buttons, text fields, etc. A View object is created from the View class and occupies a rectangular area on the screen.
+- **ViewGroup**: Subclass of the View class. Provides an invisible container that hold other Views or other ViewGroups and define their layout properties. Examples are RecyclerView, ListView, etc.
+- **Layouts**: Subclasses of the ViewGroup class. A typical layout defines the visual structure for an Android user interface. You can create it either at run time using View/ViewGroup objects or you can declare your layout using simple XML file main_layout.xml which is located in the res/layout folder of your project. Examples are LinearLayout, RelativeLayout, etc.
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:9]:
```suggestion
- **Layouts**: Subclasses of the ViewGroup class. A typical layout defines the visual structure for an Android user interface. You can create it either at run time using View/ViewGroup objects or you can declare your layout using an XML file main_layout.xml which is located in the res/layout folder of your project. Examples are LinearLayout, RelativeLayout, etc. Learn more about these example layouts below. <!-- anchor link? --> 
```

File: Articles/9_Layout_and_their_types.md [hunk 12]
```diff
@@ -51,26 +45,27 @@ Typical syntax is as follows:
...
+- Create your layouts at run time using View/ViewGroup objects or declare your layout using an XML file main_layout.xml which is located in the res/layout folder of your project.
 
-# Types of Layout
-There are numerous Layouts provided by Android which you will use in almost all the Android applications to provide different view, look and feel.
-Let's look at some of the layouts:
+## Different layout types
+
+There are numerous Layouts provided by Android which you will use in almost all the Android applications to provide different view, look and feel. Let's look at some of the layouts below.
+
+### GridView: ViewGroup to display items in a two-dimensional and scrollable grid
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:67]: reviewer note: converted to h3 and linked from above.

File: Articles/9_Layout_and_their_types.md [hunk 13]
```diff
@@ -353,54 +341,24 @@ etc...
...
-- android:paddingRight
-This is the right padding filled for the layout.
-
-- android:paddingTop
-This is the top padding filled for the layout.
 
-- android:paddingBottom
-This is the bottom padding filled for the layout.
\ No newline at end of file
+## Layout attributes
```
Comment (cadamini) [inline Articles/9_Layout_and_their_types.md:345]: table is better consumable by readers.

**Comment by utkarsh006** [comment]
@cadamini Kindly create two issues regarding both PR's and after assigning them, rest of the things would be done.

**Comment by cadamini** [comment]
@utkarsh006 done for this one, see https://github.com/utkarsh006/Eazy-Android/issues/46

**Pull request merged**